data/db-schema/src/main/resources/net/corda/db/schema/vnode-vault/migration/ledger-utxo-creation-v5.1.xml

```diff
@@ -15,6 +15,94 @@
                 tableName="utxo_transaction_output">
             <column name="type"/>
         </createIndex>
+
+    <property name="json.column.type" value="JSONB" dbms="postgresql"/>
+
+    <!-- Please note that HSQLDB schema is supported for integration test purposes only. -->
+    <property name="json.column.type" value="CLOB" dbms="hsqldb"/>
+
+    <changeSet author="R3.Corda" id="ledger-utxo-creation-v5.2">
+
+        <!-- Add new status column to utxo_transaction -->
+        <addColumn tableName="utxo_transaction">
+            <column name="status" type="VARCHAR(255)">
+                <constraints nullable="true"/>
+            </column>
+            <column name="updated" type="TIMESTAMP">
+                <constraints nullable="true"/>
+            </column>
+        </addColumn>
+
+        <!-- Migrate data from the old status table -->
+        <update tableName="utxo_transaction">
+            <column name="status"
+                    valueComputed="(select status from utxo_transaction_status uts where uts.transaction_id=id)"/>
+            <column name="updated"
+                    valueComputed="(select updated from utxo_transaction_status uts where uts.transaction_id=id)"/>
+        </update>
+
+        <!-- Drop the old status table -->
+        <dropTable tableName="utxo_transaction_status"/>
+
+        <!-- After status data migration we can make the status/updated column non-nullable -->
+        <addNotNullConstraint tableName="utxo_transaction" columnName="status" columnDataType="VARCHAR(255)"/>
+        <addNotNullConstraint tableName="utxo_transaction" columnName="updated" columnDataType="TIMESTAMP"/>
+
+        <!-- Add new custom_representation and consumed columns to utxo_transaction_output -->
+        <addColumn tableName="utxo_transaction_output">
+            <column name="custom_representation" type="${json.column.type}">
+                <constraints nullable="true"/>
+            </column>
+            <column name="consumed" type="TIMESTAMP">
+                <constraints nullable="true"/>
+            </column>
+        </addColumn>
+
+        <!--
+        Migrate data from the old visible states table
+        NOTE: We can't use <update> here because both tables have the same column naming
```

Comment:
I tried this but it seems like without somehow aliasing the base table it will not work.
It seems like the query generated by the upgrade plugin is the following:
```
SET SEARCH_PATH TO vnode_vault_37db0c9d2e17, "$user","public";
UPDATE vnode_vault_37db0c9d2e17.utxo_transaction_output SET 
	consumed = (
		SELECT vts.consumed FROM utxo_visible_transaction_state vts 
		WHERE vts.transaction_id = transaction_id                 
		AND vts.group_idx = group_idx                 
		AND vts.leaf_idx = leaf_idx
	), 
	custom_representation = (
		SELECT vts.custom_representation FROM utxo_visible_transaction_state vts 
		WHERE vts.transaction_id = transaction_id                 
		AND vts.group_idx = group_idx                 
		AND vts.leaf_idx = leaf_idx
	);
```
Executing this in the postgres DB I’m getting the following error:
```
SQL Error [21000]: ERROR: more than one row returned by a subquery used as an expression
```

I think that's because these lines:
`WHERE vts.transaction_id = transaction_id` are evaluated as `WHERE vts.transaction_id =vts.transaction_id` and returns every line.

Reply:
@vlajos ' suggestion was to create a temp view. It works like this:
```
<createView fullDefinition="false"
                    remarks="A view of utxo_visible_transaction_state with renamed fields"
                    replaceIfExists="false"
                    viewName="v_utxo_visible_transaction_state">
            SELECT
            transaction_id "vts_transaction_id",
            group_idx "vts_group_idx",
            leaf_idx "vts_leaf_idx",
            custom_representation "vts_custom_representation",
            created "vts_created",
            consumed "vts_consumed"
            FROM utxo_visible_transaction_state
        </createView>

        <update tableName="utxo_transaction_output">
            <column name="custom_representation" valueComputed="
                (SELECT vts_custom_representation
                FROM v_utxo_visible_transaction_state
                WHERE vts_transaction_id = transaction_id
                AND vts_group_idx = group_idx
                AND vts_leaf_idx = leaf_idx)
            "/>
            <column name="consumed" valueComputed="
                (SELECT vts_consumed
                FROM v_utxo_visible_transaction_state
                WHERE vts_transaction_id = transaction_id
                AND vts_group_idx = group_idx
                AND vts_leaf_idx = leaf_idx)
            "/>
        </update>

        <dropView viewName="v_utxo_visible_transaction_state"/>
```

But this seems a bit overkill for me but it might be a better solution than the raw SQL. Not sure what people think. @mbrkic-r3 @vlajos @relyafi @blsemo

Reply:
I'm no expert as I've not had to do this before, but from looking online it appears that raw SQL is the way to go? The above does seem quite complicated.

We'd presumably just need to ensure we're sticking to ANSI SQL.

Reply:
Yeah raw SQL is the best way I could find too. I just gave the view a try but not sure if it's worth it at all.